pkg/gatherers/clusterconfig/operators_pods_and_events.go

```diff
@@ -289,19 +290,14 @@ func getContainerLogs(ctx context.Context,
 			continue
 		}
 
-		// check for stack tracer
-		if found := stackTraceRegex.MatchString(logString); found {
+		logString, found := getLogWithStacktracing(logString)
```

Comment:
So do we always scan the complete log? I don't it's necessary.

Reply:
What is your suggestion?

Reply:
Let me repeat the suggestion from https://issues.redhat.com/browse/CCXDEV-4600. 
> If the last 100 lines log (the commonly gathered log) is matching the regex (there's something looking like stacktrace) then load complete log to the memory and try to record the complete stacktrace

Does it make sense?

Reply:
So, yep, but the problem in this approach is that the log gathering is already slow enough, and by adding this it will be even slower.

I agree that the result could be better, but imagine that for every pod container log we will check line by line for the stack tracing and then adding it to the final log... So, in this case, also imagine that the stack tracing is located at the first 300 log lines, but the log has 1000 lines, and we have some important info available at the end of the log.

Is it possible to happen? I don't know. Do we want to commit the rest of the log if we found a stack tracing and just focus at the stack tracing?

Reply:
No we will not check line by line for every container. The idea is to check it line by line only when the `logString` is matching the regex. I don't think it will be that slow. The assumption here is that there won't be many logs matching the stacktrace regex. Anyway it's difficult to discuss about the performance when we don't have any real numbers. 

The case you're describing (stacktrace located at the first 300 log lines) is something we won't cover. The goal is not to have every possible stacktrace in the log. The goal is IMHO to have complete stacktrace when the last 100 log lines had something looking like stacktrace (of course then there's the question what to do if we found the beginning of the stacktrace there. I guess we can't do much about it...maybe only try to gather it again, but I am not sure).

Reply:
@tremes I think we still want to limit the maximum number of fetched logs in case there was a stacktrace cuz there can be quite a lot of logs (if I got your idea correctly)

Reply:
I haven't thought about limit for the stacktrace TBH. My first thought was to get the whole stacktrace and nothing else. If we think we still need to limit it then we can probably ask ourselves if it makes sense at all (because we can't predict which part of the stacktrace will be important).

Reply:
OK. I changed things a little bit. I think that now we do have a smaller log, and also a better result to focus at the stack trace message.